lbfgs.go

```diff
@@ -1,52 +1,61 @@
+// Copyright ©2014 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
 package optimize
 
 import (
 	"github.com/gonum/floats"
 )
 
-// LBFGS implements the limited-memory BFGS algorithm. While the normal BFGS algorithm
-// makes a full approximation to the inverse hessian, LBFGS instead approximates the
-// hessian from the last Store optimization steps. The Store parameter is a tradeoff
-// between cost of the method and accuracy of the hessian approximation.
-// LBFGS has a cost (both in memory and time) of O(Store * inputDimension).
-// Since BFGS has a cost of O(inputDimension^2), LBFGS is more appropriate
-// for very large problems. This "forgetful" nature of LBFGS may also make it perform
-// better than BFGS for functions with Hessians that vary rapidly spatially.
+// LBFGS implements the limited-memory BFGS optimization method.
+//
+// While the normal BFGS algorithm makes a full approximation to the inverse
+// Hessian, LBFGS instead approximates the Hessian from the last Store
+// iterations. The Store parameter is a tradeoff between cost of the method and
```

Comment:
between storage cost. Strictly it's both, but storage is the important dimension. We can leave the last line to highlight it's both.